pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,9 +102,15 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
-		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
+		e.recorder.Eventf(config, kapi.EventTypeNormal, "Failed", "The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name)
+		// TODO: The event might not be send to server, remove the sleep when kubernetes(#27858) is resolved
+		time.Sleep(1 * time.Second)
+		return fmt.Errorf("the %s hook failed: %v, aborting deployment: %s/%s", label, err, deployment.Namespace, deployment.Name)
 	case deployapi.LifecycleHookFailurePolicyIgnore:
-		fmt.Fprintf(e.out, "warning: %s hook failed, deployment will continue: %v\n", label, err)
+		e.recorder.Eventf(config, kapi.EventTypeNormal, "Failed", "The %s hook failed: %v (ignore), deployment %s/%s will continue", label, err, deployment.Namespace, deployment.Name)
+		// TODO: The event might not be send to server, remove the sleep when kubernetes(#27858) is resolved
+		time.Sleep(1 * time.Second)
```

Comment:
@kargakis will keep this here (https://github.com/kubernetes/kubernetes/issues/27858) so we can see the event logs and will fix this as soon as the upstream issue is fixed.